### ISSUE TITLE
LanguageReverseError for downloaded subtitles

### ISSUE THREAD

**Issue by maartenbrakkee** [morpheus65535/bazarr#533]
**Describe the bug**
Downloaded subtitles don't seem to be shown in the movie view. Subtitles are on disk, but after clicking `Scan disk for subtitles` the following error appears in the log. `Encode subtitles to UTF8` is enabled. Bazarr is running as a docker container.

Message:
_BAZARR unable to index external subtitles._

Exception:
```
Traceback (most recent call last):
 File "/app/bazarr/bazarr/list_subtitles.py", line 150, in store_subtitles_movie
  only_one=settings.general.getboolean(\single_language\))
 File "/app/bazarr/bazarr/../libs/subliminal_patch/core.py", line 659, in search_external_subtitles
  only_one=only_one))
 File "/app/bazarr/bazarr/../libs/subliminal_patch/core.py", line 622, in _search_external_subtitles
  language = Language.fromietf(language_code)
 File "/app/bazarr/bazarr/../libs/subzero/language.py", line 62, in inner
  instance = f(cls, base, *args, **kwargs)
 File "/app/bazarr/bazarr/../libs/subzero/language.py", line 127, in fromietf
  return Language(*Language_.fromietf(ietf).__getstate__())
 File "/app/bazarr/bazarr/../libs/babelfish/language.py", line 123, in fromietf
  language = cls.fromalpha2(language_subtag)
 File "/app/bazarr/bazarr/../libs/babelfish/language.py", line 109, in fromcode
  return cls(*language_converters[converter].reverse(code))
 File "/app/bazarr/bazarr/../libs/babelfish/converters/__init__.py", line 165, in reverse
  raise LanguageReverseError(code)
LanguageReverseError: u\we\
```

**Expected behavior**
Subtitle should be found and shown.

**Software (please complete the following information):**
 - Bazarr: v0.8
 - OS: docker

**Comment by halali** [comment]
Can you please give us more information regarding to su titles you downloaded?
From which provider you download them, for what language. And also debug log. 
Without that  its hard to say, what is wrong.

**Comment by maartenbrakkee** [comment]
I've got the following providers enabled:
OpenSubtitles, Subscenter, Supersubtitles and TVSubtitles

Encode subtitles to UTF8 is enabled.

This specific case seems to be downloaded from OpenSubtitles.

Language:
English (non hearing-impaired and forced set to false)

Debug log:
https://pastebin.com/hhMdSuU2

**Comment by morpheus65535** [comment]
Can you show us the content of `/movies/Oldboy (2003)` directory?

**Comment by maartenbrakkee** [comment]
```
drwxrwxrwx+ 1 root    root                   86 Nov 20  2017 @eaDir
-rwxrwxrwx+ 1 Maarten            911     259464 Sep 22  2018 fanart.jpg
-rwxrwxrwx+ 1 Maarten            911      64852 Jun 12  2017 Oldboy.2003.1080p.BluRay.dut.srt
-rwxrwxrwx+ 1 Maarten            911      73245 Jun 12  2017 Oldboy.2003.1080p.BluRay.eng.srt
-rwxrwxrwx+ 1 Maarten            911      72237 Jun 12  2017 Oldboy.2003.1080p.BluRay.fin.srt
-rwxrwxrwx+ 1 Maarten            911      75878 Jun 12  2017 Oldboy.2003.1080p.BluRay.fre.srt
-rwxrwxrwx+ 1 Maarten            911 2223882791 Jun 12  2017 Oldboy.2003.1080p.BluRay.mp4
-rwxrwxrwx+ 1 Maarten            911       1266 Sep  8 18:48 Oldboy.2003.1080p.BluRay.nfo
-rwxrwxrwx+ 1 Maarten administrators      69001 Sep  9 15:17 Oldboy.2003.1080p.BluRay.srt
-rwxrwxrwx+ 1 Maarten            911      55338 Jun 12  2017 Oldboy.2003.1080p.BluRay.swe.srt
-rwxrwxrwx+ 1 Maarten            911     510331 Sep 22  2018 poster.jpg
```

**Comment by morpheus65535** [comment]
My recommandation would be to deactivate single language and redownload this subtitles. It looks like `Oldboy.2003.1080p.BluRay.srt` is recognized as an unknown language code and it cause a failure. When there's a language code in the filename, Bazarr doesn't have to guess the language inside the file.

**Comment by maartenbrakkee** [comment]
After removing all `*.srt` files in the folder and turning off the `convert to UTF-8` and `single language` options the problem seems to be gone.

However the newly downloaded subtitle (`Oldboy.2003.1080p.BluRay.en.srt`) is not displayed on the movie page. Forced refresh doesn't change a thing. When looking in the debug log I can see it is found. Any idea what this could be?

Debug log:
https://pastebin.com/uM9G00jX

**Comment by morpheus65535** [comment]
We have a couple of users having this issue but we cannot reproduce it... Probably something special in your environment. :-/

If you test with the dev branch, does the problem persist?

**Comment by maartenbrakkee** [comment]
Tested with development branch, problem persisted.

After analyzing the log I saw a wrong mapping. Seemed like the search went fine, the subtitle was found (without mapping), but is was stored with mapping.

When I removed the wrong mapping from the settings and forced refresh, the subtitle was found and shown.

I'll close this issue, but maybe some error can be added in case of subtitle not found on the location that is stored in the database?
